services/dataAccessLayer.js

```diff
@@ -0,0 +1,34 @@
+const userQuery = require("../models/users");
+
+const retrieveUsers = async ({ id = null, usernames = null, req = null }) => {
```

Comment:
this function takes three arguments, but only two of them are actually used. The req argument is never used, so it could be removed from the function signature.